src/gui/attributetable/qgsattributetablemodel.cpp

```diff
@@ -341,11 +344,14 @@ void QgsAttributeTableModel::loadAttributes()
   {
     const QgsEditorWidgetSetup setup = QgsEditorWidgetRegistry::instance()->findBest( layer(), fields[idx].name() );
     QgsEditorWidgetFactory* widgetFactory = QgsEditorWidgetRegistry::instance()->factory( setup.type() );
+    QgsFieldFormatter* fieldKit = QgsApplication::fieldKitRegistry()->fieldKit( setup.type() );
+
     if ( widgetFactory )
     {
       mWidgetFactories.append( widgetFactory );
       mWidgetConfigs.append( setup.config() );
-      mAttributeWidgetCaches.append( widgetFactory->createCache( layer(), idx, setup.config() ) );
+      mAttributeWidgetCaches.append( fieldKit->createCache( layer(), idx, setup.config() ) );
+      mFieldFormatters.append( fieldKit );
```

Comment:
That leads to trouble with python binding.
Listening to fieldFormatterRemoved would be another possibility.